closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -474,12 +576,292 @@ JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+            }
+
+            /* Identify and load the latest version from the potential libraries.
+             * This logic depends upon the order in which libnames are defined.
+             * Libraries are listed in descending order w.r.t version.
+             * Since only one library is bundled with the JDK, once any library is
+             * loaded, this is the only available and we can stop.
+             */
+            tempVersion = get_crypto_library_version(traceEnabled, result, "\t\tLoaded OpenSSL version");
+            if (tempVersion > 0) {
+                return result;
+            }
+        }
+    }
+
+    /* The attempt to load from property and OpenSSL bundled with JDK failed.
+     * Try loading the libraries in the order set out above, and retain the latest library.
+     */
+    for (i = 0; i < numOfLibs; i++) {
+        if (traceEnabled) {
+            fprintf(stdout, "Attempting to load libname from OS: %s\n", libNames[i]);
+        }
+        result = load_crypto_library(traceEnabled, libNames[i]);
+
+        if (NULL == result) {
+            continue;
+        }
+
+        /* Identify and load the latest version from the available libraries.
+         * This logic depends upon the order in which libnames are defined.
+         * It only loads the libraries which can possibly be the latest versions.
+         */
+        log_crypto_library_path(traceEnabled, result, "\tLibrary to be potentially used was loaded from");
+        tempVersion = get_crypto_library_version(traceEnabled, result, "\tLoaded OpenSSL version");
+
+        if (tempVersion <= 0) {
+            continue;
+        }
+
+        if (i >= num_of_generic) {
+            /* If library checked is not a generic one, there is no need to check further. */
+            if (previousVersion > tempVersion) {
+                unload_crypto_library(result);
+                return prevResult;
+            } else {
+                if (previousVersion != 0) {
+                    unload_crypto_library(prevResult);
+                }
+                return result;
+            }
+        } else {
+            /* If library checked is a generic one, we need to keep checking for higher versions. */
+            if (tempVersion > previousVersion) {
+                previousVersion = tempVersion;
+                prevResult = result;
+            }
+        }
+    }
+
+    /* If we reach here, it means that none of the non-generic libraries
+     * where found. However, a generic one might have been found in the
+     * process and, if so, it will be in the prevResult variable.
+     */
+    return prevResult;
+}
+
+/*
+ * Class:     jdk_crypto_jniprovider_NativeCrypto
+ * Method:    loadCrypto
+ * Signature: (ZLjava/lang/String;Ljava/lang/String;)J
+ */
+JNIEXPORT jlong JNICALL
+Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+  (JNIEnv * env, jobject jobj, jboolean traceEnabled, jstring jlibname, jstring jhomepath)
+{
+    const char *clibname = "";
```

Comment:
This local is not used (shadowed by the declaration on line 796); please remove it.

Reply:
Removed.